vllm/core/scheduler.py

```diff
@@ -826,13 +827,22 @@ def _schedule_prefills(
                 waiting_queue.popleft()
                 continue
 
+            num_lookahead_slots: int = 0
+            if self.scheduler_config.is_multi_step and enable_chunking:
+                num_lookahead_slots = self._get_num_lookahead_slots(True)
+
             # If the sequence group cannot be allocated, stop.
-            can_allocate = self.block_manager.can_allocate(seq_group)
+            # TODO (varun) : This will stop the seqs from running even if the
+            # prompt_len +  num_lookahead_slots > capcacity. This is more
+            # restrictive than what we had before.
```

Comment:
This comment is confusing. Maybe not necessary

Reply:
yeah. the relevant code is close by and that should help.